.github/workflows/nix-ci.yml

```diff
@@ -30,11 +35,46 @@ jobs:
         shell: bash
         run: nix build -L github:${{ github.repository }}/${{ github.sha }}#llvm-pretty-bc-parser.${{ matrix.ghc-version }}
 
+  build_oldGHC:
+    # This job runs the builds on older versions of GHC, verifying legacy GHC
+    # support/compatibility.
+    #
+    #  * ["legacy" v.s. "main" is primarily driven by the selection of an older
+    #    version of nixpkgs that contains the older GHC compilers.]
+    #
+    # Note that GHC 9.0 is not tested.  There are some build problems with
+    # llvm-pretty under GHC 9.0, but 9.0 was also known to produce invalid code,
+    # so it's best avoided entirely.
+    #
+    # The only difference between this job and the job above is that this job
+    # overrides the nixpkgs input on the last line below to a nixpkgs version
+    # that supports the older GHC versions.
```

Comment:
Could we instead accomplish this via configuring the `matrix` of the `build:` job (e.g., via `matrix.include`), i.e., by always passing `--override-input` but just varying which `nixpkgs` is chosen?